packages/core/src/BoltTool.ts

```diff
@@ -0,0 +1,97 @@
+import path from 'path';
+import fs from 'fs-extra';
+
+import { Tool, ToolType, Package, PackageJSON, Packages, InvalidMonorepoError } from './Tool';
+import { expandPackageGlobs, expandPackageGlobsSync } from "./expandPackageGlobs";
+
+export interface BoltPackageJSON extends PackageJSON {
+    bolt?: {
+        workspaces?: string[]
+    }
+}
+
+export const BoltTool: Tool = {
+    type: 'bolt',
+
+    async isMonorepoRoot(directory: string): Promise<boolean> {
+        try {
+            const pkgJson = (await fs.readJson(path.join(directory, "package.json"))) as BoltPackageJSON;
+            if (pkgJson.bolt && pkgJson.bolt.workspaces) {
+                return true;
+            }
+        } catch (err: any) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+        return false;
+    },
+
+    isMonorepoRootSync(directory: string): boolean {
+        try {
+            const pkgJson = fs.readJsonSync(path.join(directory, "package.json")) as BoltPackageJSON;
+            if (pkgJson.bolt && pkgJson.bolt.workspaces) {
+                return true;
+            }
+        } catch (err: any) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+        return false;
+    },
+
+    async getPackages(directory: string): Promise<Packages> {
+        const rootDir = path.resolve(directory);
```

Comment:
If we are calling `path.resolve` here then I feel like probably we should use `rootDir` over `directory` throughout this function - unless there is a strong reason to use `directory` at any given place.

This comment applies to all `getPackages(Sync)` - since they all follow the same structure